sui-execution/latest/sui-move-natives/src/test_scenario.rs

```diff
@@ -502,6 +553,149 @@ pub fn was_taken_shared(
     ))
 }
 
+// native fun was_allocated_receiving_ticket(id: ID): bool;
+pub fn was_allocated_receiving_ticket(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert!(ty_args.is_empty());
+    let id = pop_id(&mut args)?;
+    assert!(args.is_empty());
+    let object_runtime: &ObjectRuntime = context.extensions().get();
+    let was_allocated = object_runtime
+        .test_inventories
+        .allocated_tickets
+        .contains_key(&id);
+    Ok(NativeResult::ok(
+        legacy_test_cost(),
+        smallvec![Value::bool(was_allocated)],
+    ))
+}
+
+pub fn allocate_receiving_ticket_for_object(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    let ty = get_specified_ty(ty_args);
+    let id = pop_id(&mut args)?;
+
+    let Some((tag, layout, _)) = get_tag_and_layouts(context, &ty)? else {
+        return Ok(NativeResult::err(
+            context.gas_used(),
+            E_UNABLE_TO_ALLOCATE_RECEIVING_TICKET,
+        ));
+    };
+    let object_runtime: &mut ObjectRuntime = context.extensions_mut().get_mut();
+    let object_version = SequenceNumber::new();
+    let inventories = &mut object_runtime.test_inventories;
+    if inventories.allocated_tickets.contains_key(&id) {
+        return Ok(NativeResult::err(
+            context.gas_used(),
+            E_RECEIVING_TICKET_ALREADY_ALLOCATED,
+        ));
+    }
+
+    let obj_value = inventories.objects.remove(&id).unwrap();
+    let Some(bytes) = obj_value.simple_serialize(&layout) else {
+        return Ok(NativeResult::err(
+            context.gas_used(),
+            E_UNABLE_TO_ALLOCATE_RECEIVING_TICKET,
+        ));
+    };
+    let move_object = unsafe {
```

Comment:
I'm kind of surprised that you need to create this `MoveObject`/`Object` here -- I would expect the object to exist already, so that you can take it from there and wrap it in the `Receiving<T>`, is that not the case?

Reply:
The joys of test scenario...

It is sadly not the case -- the test inventories store Move values, which makes sense in every case except this one generally.

Eventually I think we should rewrite test scenario to use an actual storage layer, and simply provide hooks into that, but at least until then...